dataclients/kubernetes/ingressdefinitions.go

```diff
@@ -138,6 +139,20 @@ func (ep endpoint) targetsByServiceTarget(protocol string, serviceTarget *defini
 	return nil
 }
 
+func (ep Endpoint) targets(protocol string) []string {
+	result := make([]string, 0)
+	for _, s := range ep.Subsets {
+		for _, p := range s.Ports {
+			for _, a := range s.Addresses {
+				result = append(result, formatEndpoint(a, p, protocol))
```

Comment:
So, would we append to result every possible addr-port pair even if there is no opened port with corresponding number on corresponding address?

Reply:
yes if we would have 10 redis pods with 2 ports we would have 20 endpoints, on the other hand I would expect that redis is only having one port in the service/endpoint and for testing this is more handy, because with testcontainers you get a random port and always localhost as ip, so we need to be able to have all the ports for testing purposes.